amavisvt/client.py

```diff
@@ -489,6 +489,51 @@ def is_included(resource):
                     lambda r: '.' not in r.filename or r.filename.endswith('.')
         ]))
 
+    def api_call_check(self, checksums):
+       for api_key in self.config.apikeys:  
+           use_api_key = self.memcached.get(api_key)
+           if not use_api_key:
+               response = requests.post(self.config.api_url, {
+                'apikey': api_key,
+                              'resource': ', ' . join(checksums)
+                              }, timeout=float(self.config.timeout), headers={
+                              'User-Agent': 'amavisvt/%s (+https://ercpe.de/projects/amavisvt)' % VERSION
+                              })
+               response.raise_for_status()
+               if response.status_code == 204:
+                   #Mark api_key as not usable for 60 seconds
+                   self.memcached.set(api_key, "expired", 60)
+                   logger.info("API-Limit exceeded, API Key disabled!")
+               else:
+                   return response
+       
+       logger.info("No more API Keys available")
+       return False
+   
+    def api_call_report(self, files):
```

Comment:
This is basically the same as `api_call_check` (except for the url and the arguments), isn't it?
Please refactor it to not duplicate everything.